### DISCUSSION TITLE
Conditional width using jss

### DISCUSSION THREAD

**Discussion by just-Bri** [cssinjs/jss#1484] [General]
I'm trying to determine if there is a cleaner way to handle this:
```
style={{
  width: `${
    (suite.size / maxFloorSize) * (floorWidth - (floor.suites.length - 1) * 2)
  }px`,
}}
```  
These styles are currently applied in-line to a div inside a component.  
I think the "cleanest" way to handle this would be to turn this div in to it's own component and pass the width as a prop, and pass that to the styles. But I'm not sure if that is the best approach.

**Comment by kof** [comment]
That question can only be answered knowing what those variables are, because it would determine whos responsibility it is to make that calculation, component rendering the div or it's parent. Also I see no relationship with JSS.
